src/rdkafka_mock_handlers.c

```diff
@@ -2698,6 +2729,236 @@ static int rd_kafka_mock_handle_PushTelemetry(rd_kafka_mock_connection_t *mconn,
+        for (i = 0; i < TopicPartitionsCnt; i++) {
+                int32_t PartitionsCnt;
+                rd_kafka_Uuid_t TopicId;
+                rd_kafka_mock_topic_t *mtopic;
+                int j;
+
+                /* TopicId */
+                rd_kafka_buf_read_uuid(rkbuf, &TopicId);
+
+                mtopic = rd_kafka_mock_topic_find_by_id(mcluster, TopicId);
+
+                /* #Partitions */
+                rd_kafka_buf_read_arraycnt(rkbuf, &PartitionsCnt,
+                                           RD_KAFKAP_PARTITIONS_MAX);
+                for (j = 0; j < PartitionsCnt; j++) {
+                        int32_t Partition;
+                        rd_kafka_topic_partition_t *rktpar;
+
+                        /* Partitions[j] */
+                        rd_kafka_buf_read_i32(rkbuf, &Partition);
+
+                        if (mtopic) {
+                                rktpar = rd_kafka_topic_partition_list_add(
+                                    current_assignment, "", Partition);
+                                rd_kafka_topic_partition_set_topic_id(rktpar,
+                                                                      TopicId);
+                        }
+                }
+
+                rd_kafka_buf_skip_tags(rkbuf);
+        }
+
+        /* Inject error, if any */
+        err = rd_kafka_mock_next_request_error(mconn, resp);
+
+        if (!err) {
+                mrkb = rd_kafka_mock_cluster_get_coord(
+                    mcluster, RD_KAFKA_COORD_GROUP, &GroupId);
+
+                if (!mrkb)
+                        err = RD_KAFKA_RESP_ERR_COORDINATOR_NOT_AVAILABLE;
+                else if (mrkb != mconn->broker)
+                        err = RD_KAFKA_RESP_ERR_NOT_COORDINATOR;
+        }
+
+        if (!err) {
+                mtx_lock(&mcluster->lock);
+                mcgrp = rd_kafka_mock_cgrp_consumer_get(mcluster, &GroupId);
+                rd_assert(mcgrp);
+
+                member = rd_kafka_mock_cgrp_consumer_member_add(
+                    mcgrp, mconn, &MemberId, &InstanceId, SubscribedTopicNames,
+                    SubscribedTopicNamesCnt);
+
+                if (member) {
+                        if (MemberEpoch >= 0) {
+                                next_assignment =
+                                    rd_kafka_mock_cgrp_consumer_member_next_assignment(
+                                        member, current_assignment,
+                                        &MemberEpoch);
+                                if (MemberEpoch < 0) {
+                                        err =
+                                            RD_KAFKA_RESP_ERR_FENCED_MEMBER_EPOCH;
+                                }
+                        } else {
+                                rd_kafka_mock_cgrp_consumer_member_leave(
+                                    mcgrp, member);
+                                member = NULL;
+                        }
```

Comment:
In the static membership case (-2), the member can rejoin. We shouldn't destroy the member. You can add a FIXME for now here.

Reply:
I've implemented it and added some tests for static group membership with the mock cluster